lib/statsig/configs.ex

```diff
@@ -57,17 +66,20 @@ defmodule Statsig.Configs do
     end
   end
 
-  defp reload_configs(state) do
-    case api_client().download_config_specs(state.last_sync_time) do
+  defp reload_configs(state, timeout) do
+    case api_client().download_config_specs(state.last_sync_time, timeout) do
       {:ok, response} ->
         new_time = Map.get(response, "time", 0)
 
         if new_time >= state.last_sync_time do
-          :ets.delete_all_objects(@table_name)
-          response |> Map.get("feature_gates", []) |> update_configs(:gate)
-          response |> Map.get("dynamic_configs", []) |> update_configs(:config)
-
-          {:ok, %{state | last_sync_time: new_time}}
+          try do
+            update_specs(response)
+            {:ok, %{state | last_sync_time: new_time}}
+          rescue
+            error ->
```

Comment:
what can happen to cause this? It's a little strange to see a try/rescue in elixir code.
I would actually make it so that `update_specs` can't raise.